core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/orm.go

```diff
@@ -0,0 +1,81 @@
+package upkeepstate
+
+import (
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/lib/pq"
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type orm struct {
+	chainID *utils.Big
+	q       pg.Q
+}
+
+type persistedStateRecord struct {
+	UpkeepID            *utils.Big
+	WorkID              string
```

Comment:
?

Reply:
@amirylm we talked about this one and the idea is to keep cache size small by not storing extra data. It is extra allocations though. A bit of a tradeoff.